Build errors with gcc 8.1

On sway d6a5dfc9

```bash
[rik@T450 ~/.../github/sway]$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/8/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,fortran,objc,obj-c++,ada,go,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl --enable-libmpx --enable-offload-targets=nvptx-none --without-cuda-driver --enable-gnu-indirect-function --enable-cet --with-tune=generic --with-arch_32=i686 --build=x86_64-redhat-linux
Thread model: posix
gcc version 8.1.1 20180712 (Red Hat 8.1.1-5) (GCC) 
```

Configured with `buildtype release`:
```bash
[rik@T450 ~/.../github/sway]$ ninja -C build
ninja: Entering directory `build'
[129/259] Compiling C object 'sway/sway@@sway@exe/commands_move.c.o'.
FAILED: sway/sway@@sway@exe/commands_move.c.o 
cc -Isway/sway@@sway@exe -Isway -I../sway -Iinclude -I../include -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/uuid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/json-c -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/libdrm -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -O3 -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -DWLR_USE_UNSTABLE '-DSYSCONFDIR="//etc"' '-DSWAY_VERSION="1.0-alpha.5-165-gd6a5dfc9 (" __DATE__ ", branch '"'"'master'"'"')"' -pthread  -MD -MQ 'sway/sway@@sway@exe/commands_move.c.o' -MF 'sway/sway@@sway@exe/commands_move.c.o.d' -o 'sway/sway@@sway@exe/commands_move.c.o' -c ../sway/commands/move.c
../sway/commands/move.c: In function ‘output_in_direction’:
../sway/commands/move.c:55:21: error: ‘direction’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  enum wlr_direction direction;
                     ^~~~~~~~~
cc1: all warnings being treated as errors
[134/259] Compiling C object 'sway/sway@@sway@exe/commands_show_marks.c.o'.
ninja: build stopped: subcommand failed.
```

```bash
[rik@T450 ~/.../github/sway]$ ninja -C build
ninja: Entering directory `build'
[5/30] Compiling C object 'swaybar/swaybar@@swaybar@exe/status_line.c.o'.
FAILED: swaybar/swaybar@@swaybar@exe/status_line.c.o 
cc -Iswaybar/swaybar@@swaybar@exe -Iswaybar -I../swaybar -Iinclude -I../include -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/uuid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/json-c -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -I/usr/include/libdrm -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -O3 -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -DWLR_USE_UNSTABLE '-DSYSCONFDIR="//etc"' '-DSWAY_VERSION="1.0-alpha.5-165-gd6a5dfc9 (" __DATE__ ", branch '"'"'master'"'"')"' -pthread  -MD -MQ 'swaybar/swaybar@@swaybar@exe/status_line.c.o' -MF 'swaybar/swaybar@@swaybar@exe/status_line.c.o.d' -o 'swaybar/swaybar@@swaybar@exe/status_line.c.o' -c ../swaybar/status_line.c
In file included from ../include/swaybar/bar.h:4,
                 from ../include/swaybar/status_line.h:6,
                 from ../swaybar/status_line.c:10:
../include/pool-buffer.h:10:9: error: ‘wl_buffer’ is deprecated [-Werror=deprecated-declarations]
  struct wl_buffer *buffer;
         ^~~~~~~~~
In file included from /usr/local/include/wlr/types/wlr_box.h:13,
                 from /usr/local/include/wlr/types/wlr_output_layout.h:14,
                 from ../include/util.h:8,
                 from ../include/swaybar/config.h:6,
                 from ../swaybar/status_line.c:9:
/usr/include/wayland-server.h:73:8: note: declared here
 struct wl_buffer {
        ^~~~~~~~~
cc1: all warnings being treated as errors
[7/30] Compiling C object 'swaybar/swaybar@@swaybar@exe/i3bar.c.o'.
FAILED: swaybar/swaybar@@swaybar@exe/i3bar.c.o 
cc -Iswaybar/swaybar@@swaybar@exe -Iswaybar -I../swaybar -Iinclude -I../include -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/uuid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/json-c -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -I/usr/include/libdrm -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -O3 -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -DWLR_USE_UNSTABLE '-DSYSCONFDIR="//etc"' '-DSWAY_VERSION="1.0-alpha.5-165-gd6a5dfc9 (" __DATE__ ", branch '"'"'master'"'"')"' -pthread  -MD -MQ 'swaybar/swaybar@@swaybar@exe/i3bar.c.o' -MF 'swaybar/swaybar@@swaybar@exe/i3bar.c.o.d' -o 'swaybar/swaybar@@swaybar@exe/i3bar.c.o' -c ../swaybar/i3bar.c
In file included from ../include/swaybar/bar.h:4,
                 from ../include/swaybar/status_line.h:6,
                 from ../swaybar/i3bar.c:9:
../include/pool-buffer.h:10:9: error: ‘wl_buffer’ is deprecated [-Werror=deprecated-declarations]
  struct wl_buffer *buffer;
         ^~~~~~~~~
In file included from /usr/local/include/wlr/types/wlr_box.h:13,
                 from /usr/local/include/wlr/types/wlr_output_layout.h:14,
                 from ../include/util.h:8,
                 from ../include/swaybar/config.h:6,
                 from ../swaybar/i3bar.c:8:
/usr/include/wayland-server.h:73:8: note: declared here
 struct wl_buffer {
        ^~~~~~~~~
cc1: all warnings being treated as errors
[9/30] Compiling C object 'swaybar/swaybar@@swaybar@exe/ipc.c.o'.
FAILED: swaybar/swaybar@@swaybar@exe/ipc.c.o 
cc -Iswaybar/swaybar@@swaybar@exe -Iswaybar -I../swaybar -Iinclude -I../include -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/uuid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/json-c -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -I/usr/include/libdrm -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -O3 -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -DWLR_USE_UNSTABLE '-DSYSCONFDIR="//etc"' '-DSWAY_VERSION="1.0-alpha.5-165-gd6a5dfc9 (" __DATE__ ", branch '"'"'master'"'"')"' -pthread  -MD -MQ 'swaybar/swaybar@@swaybar@exe/ipc.c.o' -MF 'swaybar/swaybar@@swaybar@exe/ipc.c.o.d' -o 'swaybar/swaybar@@swaybar@exe/ipc.c.o' -c ../swaybar/ipc.c
In file included from ../include/swaybar/bar.h:4,
                 from ../include/swaybar/ipc.h:4,
                 from ../swaybar/ipc.c:8:
../include/pool-buffer.h:10:9: error: ‘wl_buffer’ is deprecated [-Werror=deprecated-declarations]
  struct wl_buffer *buffer;
         ^~~~~~~~~
In file included from /usr/local/include/wlr/types/wlr_box.h:13,
                 from /usr/local/include/wlr/types/wlr_output_layout.h:14,
                 from ../include/util.h:8,
                 from ../include/swaybar/config.h:6,
                 from ../swaybar/ipc.c:7:
/usr/include/wayland-server.h:73:8: note: declared here
 struct wl_buffer {
        ^~~~~~~~~
cc1: all warnings being treated as errors
[10/30] Compiling C object 'swaybar/swaybar@@swaybar@exe/bar.c.o'.
FAILED: swaybar/swaybar@@swaybar@exe/bar.c.o 
cc -Iswaybar/swaybar@@swaybar@exe -Iswaybar -I../swaybar -Iinclude -I../include -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/uuid -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/json-c -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -I/usr/include/libdrm -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -O3 -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -DWLR_USE_UNSTABLE '-DSYSCONFDIR="//etc"' '-DSWAY_VERSION="1.0-alpha.5-165-gd6a5dfc9 (" __DATE__ ", branch '"'"'master'"'"')"' -pthread  -MD -MQ 'swaybar/swaybar@@swaybar@exe/bar.c.o' -MF 'swaybar/swaybar@@swaybar@exe/bar.c.o.d' -o 'swaybar/swaybar@@swaybar@exe/bar.c.o' -c ../swaybar/bar.c
In file included from ../include/swaybar/bar.h:4,
                 from ../include/swaybar/status_line.h:6,
                 from ../swaybar/bar.c:22:
../include/pool-buffer.h:10:9: error: ‘wl_buffer’ is deprecated [-Werror=deprecated-declarations]
  struct wl_buffer *buffer;
         ^~~~~~~~~
In file included from /usr/local/include/wlr/types/wlr_box.h:13,
                 from /usr/local/include/wlr/types/wlr_output_layout.h:14,
                 from ../include/util.h:8,
                 from ../include/swaybar/config.h:6,
                 from ../swaybar/bar.c:20:
/usr/include/wayland-server.h:73:8: note: declared here
 struct wl_buffer {
        ^~~~~~~~~
cc1: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```

--- a/meson.build
+++ b/meson.build
@@ -12,6 +12,7 @@ project(
 add_project_arguments('-Wno-unused-parameter', language: 'c')
 add_project_arguments('-Wno-unused-function', language: 'c')
 add_project_arguments('-Wno-unused-result', language: 'c')
+add_project_arguments('-DWL_HIDE_DEPRECATED', language: 'c')
 add_project_arguments('-DWLR_USE_UNSTABLE', language: 'c')
 
 cc = meson.get_compiler('c')

--- a/sway/commands/move.c
+++ b/sway/commands/move.c
@@ -52,7 +52,7 @@ static struct sway_container *output_in_direction(const char *direction_string,
 		{ "right", WLR_DIRECTION_RIGHT },
 	};
 
-	enum wlr_direction direction;
+	enum wlr_direction direction = 0;
 
 	for (size_t i = 0; i < sizeof(names) / sizeof(names[0]); ++i) {
 		if (strcasecmp(names[i].name, direction_string) == 0) {

